lib/vm/src/vmcontext.rs

```diff
@@ -421,14 +422,42 @@ mod test_vmtable_definition {
     }
 }
 
+/// A typesafe wrapper around the storage for a global variables.
+///
+/// # Safety
+///
+/// Accessing the different members of this union is always safe because there
+/// are no invalid values for any of the types and the whole object is
+/// initialized by VMGlobalDefinition::new().
+#[derive(Clone, Copy)]
+#[repr(C, align(16))]
+pub union VMGlobalDefinitionStorage {
+    as_i32: i32,
+    as_u32: u32,
+    as_f32: f32,
+    as_i64: i64,
+    as_u64: u64,
+    as_f64: f64,
+    as_u128: u128,
+    bytes: [u8; 16],
+}
```

Comment:
Seems like a reasonable change!  The one thing I'd mention is that I believe Wasm numbers are always Little Endian, I'd suspect this to do the wrong thing on a Big Endian machine

Reply:
I think the intention is that the caller always use the correct accessor type, like as_i32() / as_i32_mut() if the wasm type of the global is I32. On the other hand, it's not rust-unsafe to switch up the types, it just may produce a surprising value.

Reply:
It's not strictly unsafe, but yeah, as you said, it produces surprising values if the type isn't correctly selected.